### DISCUSSION TITLE
TypeError:: How to write Controller generic without pass control props ?

### DISCUSSION THREAD

**Discussion by genshinw** [react-hook-form/react-hook-form#8748] [Q&A] [answered]
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/12270447/181204595-56a81088-7f6c-4dde-8a08-58abf489bd7f.png">
The value type not right. How to write Controller generic without pass control props with FormProvider?

**Comment by genshinw** [comment, marked as answer]
I found two solutions.

1.  `useFormContext<IKycEnterpriseForm>` to get the 'control' value and pass to the `Controller ` explicitly.
2. `<Controller<IKycEnterpriseForm, "fieldName">  name="fieldName" />`, I thought It looks strange.

Is there any other better solution？

**Reply by Moshyfawn** [reply]
Providing your `FormValues` generic to `useFormContext` is the recommended way of RHF.
